bot/core/events/registry.py

```
@@ -0,0 +1,467 @@
+        return EventPriority.NORMAL
+
+
+class EventFilter(ABC):
+    """
+    Abstract base class for event filters.
+
+    Event filters can modify or reject events before they are processed by handlers.
+    """
+
+    @abstractmethod
+    def filter_event(self, event: Event) -> Optional[Event]:
```

Ruff (0.8.2)

Undefined name `Optional`

(F821)